src/Neo/SmartContract/Native/ContractManagement.cs

```diff
@@ -110,7 +110,7 @@ internal override async ContractTask OnPersistAsync(ApplicationEngine engine)
                     }
 
                     // Emit native contract notification
-                    engine.SendNotification(Hash, state is null ? "Deploy" : "Update", new VM.Types.Array(engine.ReferenceCounter) { contract.Hash.ToArray() });
```

Comment:
We should avoid these changes because we will need to do the old way before the HF

Reply:
@Jim8y just allow null as rc

Reply:
> We should avoid these changes because we will need to do the old way before the HF

@shargon I would love to see you keep fixing ur solution, I found that neogo style of rc management is different from c# and requires update that is hard to do a hf. For instance, it manage all gc in stack, while c# do it in both stack and item....making the gc inc and dec process different.

Reply:
But we will need to execute the previous blocks as before, isn't it?

Reply:
> But we will need to execute the previous blocks as before, isn't it?

I understand, that is why i suggest you to keep working on your solution, this pr migrate complete neogo gc solution, but it requires many changes that is hard to place a hardfork there. maybe hardfork is not needed at all? since neogo works fine all the time.... and this pr copies that solution, should work the same (expected) as neogo.

Reply:
Just allow null reference counters, and create new methods in the interface for the new calls, and is compatible

Reply:
good idea, let me have a try.